.github/workflows/build_and_test.yml

```diff
@@ -408,6 +408,14 @@ jobs:
         key: pyspark-coursier-${{ hashFiles('**/pom.xml', '**/plugins.sbt') }}
         restore-keys: |
           pyspark-coursier-
+    - name: Free up disk space
+      run: |
+        dpkg-query -Wf '${Installed-Size}\t${Package}\n' | sort -n | tail -n 100
+        df -h
+        rm -rf /__t/CodeQL || echo "fail to delete /__t/CodeQL"
+        rm -rf /__t/go || echo "fail to delete /__t/go"
+        rm -rf /__t/node || echo "fail to delete /__t/node"
```

Comment:
deleting those directories in `Dockerfile` takes no effect ...

Reply:
```
88M	/__e/node12
55M	/__e/node12_alpine
101M	/__e/node16
89M	/__e/node16_alpine
8.2G	/__t/CodeQL
16K	/__t/Java_Temurin-Hotspot_jdk
487M	/__t/PyPy
1.2G	/__t/Python
62M	/__t/Ruby
1.2G	/__t/go
379M	/__t/node
16K	/__w/_PipelineMapping
26M	/__w/_actions
68K	/__w/_temp
681M	/__w/spark
```

Reply:
I can not find an official way to uninstall [CodeQL](https://codeql.github.com/docs/codeql-overview/about-codeql/#:~:text=In%20CodeQL%2C%20code%20is%20treated%20like%20data.%20Security,write%20your%20own%20to%20use%20in%20custom%20analyses.) (like pip/apt-get/etc)

Reply:
cc @Yikun do you happen to know can we remove `/__t/CodeQL` in this way?

Reply:
TBH, I don't find a good way to remove CodeQL/Go/Node

also cc @LuciferYang @HyukjinKwon

Reply:
According the link:
https://github.com/actions/runner-images/blob/main/images/linux/Ubuntu2204-Readme.md#installed-apt-packages

seems only above package can be unintall by apt, for other one, we can only cleanup by this way, so this PR seems a okay way.

Reply:
is `CodeQL` installed by the `Ubuntu 22.04` runner?
I am surprised that I don't find it in a non-container job (in https://github.com/apache/spark/pull/42241)